src/material-experimental/mdc-dialog/testing/dialog-opener.spec.ts

```diff
@@ -0,0 +1,81 @@
+import {Component, Inject} from '@angular/core';
+import {fakeAsync, TestBed, flush} from '@angular/core/testing';
+import {
+  MatTestDialogOpenerModule,
+  MatTestDialogOpener,
+} from '@angular/material-experimental/mdc-dialog/testing';
+import {
+  MAT_DIALOG_DATA,
+  MatDialogRef,
+  MatDialogState,
+} from '@angular/material-experimental/mdc-dialog';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+
+describe('MDC-based MatTestDialogOpener', () => {
+  beforeEach(fakeAsync(() => {
+    TestBed.configureTestingModule({
+      imports: [MatTestDialogOpenerModule, NoopAnimationsModule],
+      declarations: [ExampleComponent],
+    });
+
+    TestBed.compileComponents();
+  }));
+
+  it('should open a dialog when created', fakeAsync(() => {
+    const fixture = TestBed.createComponent(MatTestDialogOpener.withComponent(ExampleComponent));
+    flush();
+    expect(fixture.componentInstance.dialogRef.getState()).toBe(MatDialogState.OPEN);
+    expect(document.querySelector('mat-dialog-container')).toBeTruthy();
+  }));
+
+  it('should throw an error if no dialog component is provided', () => {
+    expect(() => TestBed.createComponent(MatTestDialogOpener)).toThrow(
+      Error('MatTestDialogOpener does not have a component provided.'),
+    );
+  });
+
+  it('should pass data to the component', fakeAsync(() => {
+    const config = {data: 'test'};
+    TestBed.createComponent(MatTestDialogOpener.withComponent(ExampleComponent, config));
+    flush();
+    const dialogContainer = document.querySelector('mat-dialog-container');
+    expect(dialogContainer!.innerHTML).toContain('Data: test');
+  }));
+
+  it('should get closed result data', fakeAsync(() => {
+    const config = {data: 'test'};
+    const fixture = TestBed.createComponent(
+      MatTestDialogOpener.withComponent<ExampleComponent, ExampleDialogResult>(
+        ExampleComponent,
```

Comment:
I was thinking we could at least automatically infer the `ExampleComponent`, saving on one type parameter.